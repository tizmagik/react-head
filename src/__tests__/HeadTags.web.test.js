import React from 'react';
import PropTypes from 'prop-types';
import { mount } from 'enzyme';

const ReactDOMMock = {
  createPortal: jest.fn(() => 'createPortal'),
};
jest.setMock('react-dom', ReactDOMMock);
const removeMock = jest.fn();
const qsMock = jest.fn(() => ({
  remove: removeMock,
}));
document.head.querySelector = qsMock;

describe('HeadTag during client rendering', () => {
  // eslint-disable-next-line global-require
  const HeadTag = require('../HeadTag').default;

  const Wrapper = ({ children }) => <div>{children}</div>; // eslint-disable-line react/prop-types
  Wrapper.contextTypes = {
    reactHeadTags: PropTypes.object,
  };
  const arr = [];
  mount(
    <Wrapper>
      Yes render
      <HeadTag tag="meta" name="x" content="testing" />
      <HeadTag tag="title">Test title</HeadTag>
    </Wrapper>,
    {
      context: {
        reactHeadTags: {
          add: c => arr.push(c),
        },
      },
    }
  );

  it('removes head tags added during ssr', () => {
    expect(qsMock).toHaveBeenCalledWith('meta[name="x"][content="testing"][data-rh=""]');
    expect(qsMock).toHaveBeenCalledWith('title[data-rh=""]');
    expect(removeMock).toHaveBeenCalledTimes(2);
  });
  it('renders into document.head portal', () => {
    expect(ReactDOMMock.createPortal).toHaveBeenCalledTimes(2);
    expect(ReactDOMMock.createPortal).toHaveBeenCalledWith(expect.any(Object), document.head);
  });
});
