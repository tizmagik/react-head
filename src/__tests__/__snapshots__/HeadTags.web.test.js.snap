// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadTag during client mounts and unmounts title 1`] = `
Array [
  <title>
    Static
  </title>,
  <button
    onClick={[Function]}
  >
    toggle
  </button>,
]
`;

exports[`HeadTag during client mounts and unmounts title 2`] = `
Array [
  <title>
    Dynamic
  </title>,
  <button
    onClick={[Function]}
  >
    toggle
  </button>,
]
`;

exports[`HeadTag during client mounts and unmounts title 3`] = `
Array [
  <title>
    Static
  </title>,
  <button
    onClick={[Function]}
  >
    toggle
  </button>,
]
`;

exports[`HeadTag during client renders only the last meta with the same name 1`] = `
Array [
  <meta>
    Static 1
  </meta>,
  <meta
    name="name1"
  >
    Static 2
  </meta>,
  <button
    onClick={[Function]}
  >
    toggle
  </button>,
  <button
    onClick={[Function]}
  >
    toggle
  </button>,
]
`;

exports[`HeadTag during client renders only the last meta with the same name 2`] = `
Array [
  <meta>
    Static 1
  </meta>,
  <meta
    name="name1"
  >
    Dynamic 1
  </meta>,
  <button
    onClick={[Function]}
  >
    toggle
  </button>,
  <button
    onClick={[Function]}
  >
    toggle
  </button>,
]
`;

exports[`HeadTag during client renders only the last meta with the same name 3`] = `
Array [
  <meta
    name="name1"
  >
    Dynamic 1
  </meta>,
  <button
    onClick={[Function]}
  >
    toggle
  </button>,
  <meta>
    Dynamic 2
  </meta>,
  <button
    onClick={[Function]}
  >
    toggle
  </button>,
]
`;

exports[`HeadTag during client renders only the last meta with the same name 4`] = `
Array [
  <meta>
    Static 1
  </meta>,
  <meta
    name="name1"
  >
    Dynamic 1
  </meta>,
  <button
    onClick={[Function]}
  >
    toggle
  </button>,
  <button
    onClick={[Function]}
  >
    toggle
  </button>,
]
`;

exports[`HeadTag during client renders only the last meta with the same name 5`] = `
Array [
  <meta>
    Static 1
  </meta>,
  <meta
    name="name1"
  >
    Static 2
  </meta>,
  <button
    onClick={[Function]}
  >
    toggle
  </button>,
  <button
    onClick={[Function]}
  >
    toggle
  </button>,
]
`;

exports[`HeadTag during client renders only the last title 1`] = `
Array [
  <div />,
  <div />,
  <div>
    <title>
      Title 3
    </title>
  </div>,
]
`;

exports[`HeadTag during client switches between titles 1`] = `
Array [
  <title>
    Static
  </title>,
  <button
    onClick={[Function]}
  >
    0
  </button>,
  <button
    onClick={[Function]}
  >
    1
  </button>,
]
`;

exports[`HeadTag during client switches between titles 2`] = `
Array [
  <title>
    Title 1
  </title>,
  <button
    onClick={[Function]}
  >
    0
  </button>,
  <button
    onClick={[Function]}
  >
    1
  </button>,
]
`;

exports[`HeadTag during client switches between titles 3`] = `
Array [
  <button
    onClick={[Function]}
  >
    0
  </button>,
  <title>
    Title 2
  </title>,
  <button
    onClick={[Function]}
  >
    1
  </button>,
]
`;

exports[`HeadTag during client switches between titles 4`] = `
Array [
  <title>
    Title 1
  </title>,
  <button
    onClick={[Function]}
  >
    0
  </button>,
  <button
    onClick={[Function]}
  >
    1
  </button>,
]
`;
